testing/workflows/components/kfctl_go_test.jsonnet

```diff
@@ -0,0 +1,372 @@
+        // This way we can configure it on a single step and reuse it on subsequent steps.
+        // The directory should be unique for each workflow so that multiple workflows don't collide.
+        name: "KUBECONFIG",
+        value: kubeConfig,
+      },
+    ] + prowEnv + env_vars,
+    volumeMounts: [
+      {
+        name: dataVolume,
+        mountPath: mountPath,
+      },
+      {
+        name: "github-token",
+        mountPath: "/secret/github-token",
+      },
+      {
+        name: "gcp-credentials",
+        mountPath: "/secret/gcp-credentials",
+      },
+    ],
+  },
+  metadata: {
+    labels: prowDict {
+      workflow: params.name,
+      workflow_template: workflow_template,
+      step_name: step_name,
+    },
+  },
+  sidecars: sidecars,
+};  // buildTemplate
+
+local componentTests = util.kfTests {
+  name: "gke-tests",
+  platform: "gke",
+  testDir: testDir,
+  kubeConfig: kubeConfig,
+  image: image,
+  workflowName: params.workflowName,
+  buildTemplate+: {
+    argoTemplate+: {
+      container+: {
+        metadata+: {
+          labels: prowDict {
+            workflow: params.name,
+            workflow_template: workflow_template,
+          },
+        },
+      },
+    },
+  },
+};
+
+// Create a list of dictionary.c
+// Each item is a dictionary describing one step in the graph.
+local dagTemplates = [
+  {
+    template: buildTemplate("checkout",
+                            ["/usr/local/bin/checkout.sh", srcRootDir],
+                            env_vars=[{
+                              name: "EXTRA_REPOS",
+                              value: "kubeflow/tf-operator@HEAD;kubeflow/testing@HEAD",
+                            }]),
+    dependencies: null,
+  },  // checkout
+  {
+    template: buildTemplate("create-pr-symlink", [
+      "python",
+      "-m",
+      "kubeflow.testing.prow_artifacts",
+      "--artifacts_dir=" + outputDir,
+      "create_pr_symlink",
+      "--bucket=" + bucket,
+    ]),  // create-pr-symlink
+    dependencies: ["checkout"],
+  },  // create-pr-symlink
+  {
+    template: buildTemplate(
+      "build-kfctl",
+      [        
+        "pytest",
+        "kfctl_go_test.py",
+        // I think -s mean stdout/stderr will print out to aid in debugging.
+        // Failures still appear to be captured and stored in the junit file.
+        "-s",
+        // Increase the log level so that info level log statements show up.
+        "--log-cli-level=info",
+        // Test timeout in seconds.
+        "--timeout=500",
+        "--junitxml=" + artifactsDir + "/junit_kfctl-build-test.xml",
+      ],
+      working_dir=srcDir+ "/testing/kfctl",
+    ),
+    dependencies: ["checkout"],
+  },
+];
+
+// Each item is a dictionary describing one step in the graph
+// to execute on exit
+local deleteKubeflow = util.toBool(params.deleteKubeflow);
+
+local deleteStep = if deleteKubeflow then
+  [{
+    template: buildTemplate(
```

Comment:
Seems previous step just build the binary, didn't actually deploy a cluster.
Do we need delete step here?

Reply:
Done.